### ISSUE TITLE
Google Contact is deleted if created in TB and synced twice

### ISSUE THREAD

**Issue by davidm3i** [jobisoft/DAV-4-TbSync#207]
## Your environment

TbSync version: 2.11
Thunderbird version: 68.6.0 (32-Bit) (on Windows 10)

[X] Yes, I have installed the latest available (beta) version from 
https://github.com/jobisoft/TbSync/releases 
and my issue is not yet fixed, I can still reproduce it.

## Expected behavior
Creating a contact in TB on PC and syncing makes it available in Google Contacts. Syncing multiple times keeps the contact in the cloud and in TB.

## Actual behavior
Creating a contact in TB on PC and syncing makes it available in Google Contacts. (works) Syncing another time deletes it everywhere.

## Steps to reproduce
1. Create a Google contact in TB on PC with Windows 10 and sync the adress book.
2. Click "sync now" a second time.
3. Contact is deleted on every device and in the cloud.

Remark: If the contact is created on another device, everything works fine and nothing gets deleted.

To help resolving your issue, enable debug logging (TbSync Account Manager -> Help) and send me the debug.log via e-mail (use the title of your issue as subject of the email).

**Comment by jobisoft** [comment]
I do not have a google device, but used "Online Google Contacts" to add the phone number to the contact created in TB and that phone number showed up in TB just fine. The contact still exists in "Online Google Contacts".

Do you observe your behaviour only if a google device is involved, or also when you use the "Online Google Contacts" to modify the contact?

**Comment by jobisoft** [comment]
Also please switch to my beta release channel so you use the latest development state:
https://tbsync.jobisoft.de

**Comment by davidm3i** [comment]
Ok, so I did some further testing:
Forget about the Android App. The whole thing seems to be something local as I am not able to create a contact in TB and keep it: On first sync it shows up in the cloud, on the second sync (without modification anywhere) it gets deleted. I changed to your beta and everything stays the same. I created an empty profile in TB where I just try to sync the contacts.
Do you have an idea what might be going wrong?

**Comment by jobisoft** [comment]
can you remove TbSync and add the google calendar manually to lightning and see if the behavior changes?

**Comment by jobisoft** [comment]
Wrong issue, forget my last comment

**Comment by jobisoft** [comment]
Now back to your issue: I have no idea what is going on :-( I am on Windows 10 myself and cannot reproduce that here. Can you please enable debug logging with highest log details under help in the tbsync account manager, restart TB and than repeat those steps? Afterwards send me the log so I could have a look at what happens there.

**Comment by davidm3i** [comment]
Hi, so I did some debugging myself and it had something to do with an error related to my Google account. Creating a new one and syncing it worked absolutely flawless. I assume it was caused by gContactSync which I tried out before I found your Add-on. I switched since it frequently messed up my contacts.
Therefore, I went through the hustle to delete and recreate my account and everything works fine now as far as I can see. Feel free to delete this issue. Sorry for the time investment necessary from your side. Should have spent more time on investigating myself before opening up an issue.
Well done Add-on so far! :)

For reference in case somebody has a similar issue: The debug log showed the following soft error in between:
[EventLog] : softerror::404
URL:
https://www.googleapis.com/carddav/v1/principals/XXX@gmail.com/lists/default/00f0000a00ff0000 (DELETE)

Request:

Response:
{
  "error": {
    "code": 404,
    "message": "Requested entity was not found.",
    "status": "NOT_FOUND"
  }
}

I changed the URL for privacy reasons.
